### ISSUE TITLE
There is a print('abc') somewhere when executing callbacks.Gap

### ISSUE THREAD

**Issue by igres26** [Quantum-TII/qibo#224]
When I do an adiabatic evolution with a `callback.Gap` it prints `abc` over and over again.

When I do an adiabatic evolution without callbacks enabled, the printing stops.

Something similar happens when comparing matrices with

```python
np.testing.assert_allclose(H0T.dense.matrix, H0T2.dense.matrix)
```

so it might be in the `.dense.matrix` module. Also, when running this last line twice on a jupyter notebook, the second time the `abc` print disappears

**Comment by scarrazza** [comment]
Thanks, here is the line:

https://github.com/Quantum-TII/qibo/blob/63fbe5c5f770a7e3f5e9c8ff75df915585e3ba25/src/qibo/base/hamiltonians.py#L324

**Comment by stavros11** [comment]
Yes, I also noticed that when I was doing the tests for the Trotter Hamiltonian. It is a leftover from some "debugging" I was doing and I forgot to remove it (sorry!)

**Comment by scarrazza** [comment]
I suggest to fix that directly in #216.

**Comment by stavros11** [comment]
> I suggest to fix that directly in #216.

I agree. Just fixed it there.
